addons/web/static/tests/views/fields/reference_field_tests.js

```diff
@@ -525,6 +527,54 @@ QUnit.module("Fields", (hooks) => {
         );
     });
 
+    QUnit.test("Many2One 'Search More...' updates on resModel change", async function (assert) {
```

Comment:
Your test look complexe. You should check if the right value is display. And don't use var

Reply:
Which value are you referring to?
The test checks if the "Search more..." option is working properly, which means whether `props.resModel` or `props.relation` is properly assigned. Do you want me to verify the values displayed in the modal instead?